### ISSUE TITLE
User manual?

### ISSUE THREAD

**Issue by BurnerCorvus** [davidson16807/tectonics.js#36]

**Comment by BurnerCorvus** [comment]
I was brought to tectonics.js because of a mention in wasd20's YouTube channel.  I love the idea of a world built on some simulated geology rather than fractals that look good.  I am, however getting confused on a sense of scale and some of the features.

I assume the world modeled is the size of Earth but what is the size of the triangles it looks to be the minimum resolution?  What heights do the colors of the topographic represent?  the colors for air temperature?

Plant productivity shows just three planet-circling green bands separated by two narrower white bands plus the poles.  Other views such as air pressure, precipitation, and plates don't seem to do anything at all, and when I try to export, I get the headers but no data in the .csv file.  Are these future enhancements.

In any event, thank you for what looks to be a lot of work so far.

**Comment by davidson16807** [comment]
Agreed. Let's get a list together for objectives. It might be easier to put this together once we have an idea what it looks like. 

I'm picturing we expand what's hosted on the github wiki [here](https://github.com/davidson16807/tectonics.js/wiki). Documentation should be broken into two categories, one for users and one for developers. What does user documentation look like for you? What do you think its structure looks like? 

BTW, If anyone wants to mess around a bit with the model and post their findings on the wiki I guarantee I won't mind. Right now any documentation is better than no documentation and I'm not going to judge if things get messy.

**Comment by BurnerCorvus** [comment]
Burner Corvus here.

I can help with the user documentation; I don't know squat about coding and my math ran aground at Trig. :-)

I'd think a full blown Wiki would be overdoing it, a couple pages would probably cover everything a user would need to know.

Let's start with the basics: How many cells are simulated? From that, the area per cell could be calculated for various-size planets.

It is looking like the time passage is variable from 1yr to 1myr per second. How many calculations are made per second?

**Comment by davidson16807** [comment]
>I can help with the user documentation; I don't know squat about coding and my math ran aground at Trig. :-)

Awesome! I appreciate it very much! No worries about background. It's probably better having a layman write it anyways since that's who the audience will be. :smiley:  

>I'd think a full blown Wiki would be overdoing it, a couple pages would probably cover everything a user would need to know.

Alright. I'll see what comes out of this and if the wiki looks like a good place for it I'll put it there myself. For right now, let's just write something and not worry about formats. 

To answer your questions:

>How many cells are simulated?

10242, that's [50,000km](https://www.wolframalpha.com/input/?i=surface+area+of+the+earth+%2F10242) for an earth-sized planet. 

You can adjust that number in factors of 4 by setting a query string in the url, i.e. type http://davidson16807.github.io/tectonics.js/?resolution=6 in the address bar and hit enter. The "?resolution=6" part means it will run the simulation at resolution level 6, which has 40962 cells. This is currently the highest number of cells you can run the simulation with. 

>It is looking like the time passage is variable from 1yr to 1myr per second. 

Time passage can actually slower than that. The slowest you can go is to run it in real time, or 1 second per second. 

>How many calculations are made per second?

Depends on what you mean by "calculation". I'm going to assume you mean a time step. The model runs a single timestep every frame, and there's usually 20-60 frames per second on my machine. It automatically adjusts the size of the timestep to maintain the correct passage of time.

**Comment by davidson16807** [comment]
And almost forgot your questions from earlier:

>I assume the world modeled is the size of Earth

Correct, though that will probably change in the future. 

>What heights do the colors of the topographic represent?  the colors for air temperature?

Unfortunately, these colors scale dynamically so it depends on the world you're rendering. I've wanted for some time to add a color legend and should probably create a new issue to track that.

>Plant productivity shows just three planet-circling green bands separated by two narrower white bands plus the poles. 

Plant productivity relies on precipitation, which has a pretty simplistic model right now. I intend to fix that in the future. 

>Other views such as air pressure, precipitation, and plates don't seem to do anything at all

those views were broken and I've since fixed them as part of issue #35. Not sure about the Plates view, it seems to be working for me. Is it broken for you? What does it look like?

> and when I try to export, I get the headers but no data in the .csv file. Are these future enhancements.

I'm not able to replicate this. What browser are you on? Are there any steps to reproduce this?

**Comment by BurnerCorvus** [comment]
Aha! Figured out I need to return to satellite view each time before selecting one of the other views. Still no luck on the export, though. Attached are a screenshot (with an ancient copy of Excel) and the .csv file itself. In Firefox I just clicked on Export and CSV, the only option, then Open or Save.

**Comment by BurnerCorvus** [comment]
Aha!  I figured out I generally have to return to Satellite View before selecting one of the other views, not jump from view to view; Plates comes up fine that way.

For exporting, in Firefox I click on Export, CSV (the only option), then Open (with an ancient version of Excel or the free version of Open Office) or Save.  Attached are the .csv file and a screenshot.

**Comment by davidson16807** [comment]
Could you try uploading that csv/screenshot again? I don't see it anywhere.

**Comment by BurnerCorvus** [comment]
Second try for the screenshot, along with the .csv file itself.

I tried running the program in Edge to see if it was somehow browser specific, but it doesn't work at all.  The Tectonics.js header comes up but Run is missing down below; just About Blog Credits and Code show.  Clicking Here on the About page does not work either.  I suspect some sort of permission is needed but haven't explored what it might be.

**Comment by davidson16807** [comment]
Still nothing, could you try posting on imgur, or maybe sending it through email? 

About Edge: I don't regularly test on windows. Knowing Microsoft, I imagine they're trying to play games with the browser market as if it's still 1996. I'm not surprised but I'll check it out.

**Comment by BurnerCorvus** [comment]
I was attaching the files to what I thought was a regular 'ol email but on actually checking, I can see that I've been replying to tectonics.js@noreply.github.com Lets's try a direct link bypassing GitHub all together. My address is besterson@cox.net Send to that and I'll try replying to that instead of relying on GH.

**Comment by BurnerCorvus** [comment]
[Headsmack] Or pick them up here:
http://www.fileconvoy.com/dfl.php?id=g28ebb083b4d12461100015484473d648198f2a42bc

I don't have any photo-sharing accounts set up.

**Comment by davidson16807** [comment]
Awesome! that last one did the trick.

I see what you mean and I'll go check it out.

**Comment by davidson16807** [comment]
Seems to be related to [this](https://stackoverflow.com/questions/48589592/source-map-error-typeerror-networkerror-when-attempting-to-fetch-resource-o) issue. I'm going to change it to use a newer technique that's already used for .sim file downloads. I'm going to release a fix to prod soon.

**Comment by davidson16807** [comment]
There's a fix out in prod right now. I setup issue #40 to track this. Go check it out and let me know if that fixes it.

**Comment by davidson16807** [comment]
As for the Edge issue: I setup issue #41 to track this. It's since been fixed.

**Comment by TheNewParad1gm** [comment]
I'd really love a manual, or help file that set out what the various settings do. 

I find seed pretty confusing as I've no idea what it is 'seeding'... it also doesn't change when you hit randomize, which i'd expect it too, nor does it seem to do anything when you type a new seed in. It'd be handy to know what seed actually impacted.

I see elsewherethat colours for temp, density, etc are adaptive, but it'd be useful to know which was 'hottest' and 'coldest' for example (I realise temp is fairly obvious, but not everything is).

**Comment by TheNewParad1gm** [comment]
Any chance of some advice on how seed works? 

Changing it manually doesn't seem to do anything, nor does the seed no. change when you click randomize. Not sure what the purpose of "copy seed" is given the above. The number does change each time you refresh the page (or change the resolution) so I'm clearly missing something.

(running in Chrome)
